src/main/java/bio/terra/service/snapshot/SnapshotRequestValidator.java

```diff
@@ -72,13 +73,22 @@ private void validateSnapshotContents(
               case BYROWID:
                 validateSnapshotRowIdSpec(contents.getRowIdSpec(), errors);
                 break;
+              case BYREQUESTID:
+                validateSnapshotRequestIdSpec(contents.getRequestIdSpec(), errors);
+                break;
               default:
                 errors.rejectValue("contents", "SnapshotContentsModeInvalid");
             }
           });
     }
   }
 
+  private void validateSnapshotRequestIdSpec(SnapshotRequestIdModel requestIdSpec, Errors errors) {
+    if (requestIdSpec == null) {
+      errors.rejectValue("contents", "SnapshotRequestIdMissing");
+    }
```

Comment:
Probably not worth doing now, but there's a lot of duplication here that could be improved. One approach
```
  <T> void rejectIfNull(Supplier<T> value, String field, String errorCode, Errors errors) {
    if (value.get() == null) {
      errors.rejectValue(field, errorCode);
    }
  }

  private void validateSnapshotRequestIdSpec(SnapshotRequestContentsModel contents, Errors errors) {
    rejectIfNull(contents::getRequestIdSpec, "contents", "SnapshotRequestIdMissing", errors);
    }
  }
```